src/python/grpcio/grpc/_server.py

```diff
@@ -560,7 +570,22 @@ def _handle_unrecognized_method(rpc_event):
     return rpc_state
 
 
-def _handle_with_method_handler(rpc_event, method_handler, thread_pool):
+def _reject_call(rpc_event):
+    operations = (cygrpc.operation_send_initial_metadata(_common.EMPTY_METADATA,
+                                                         _EMPTY_FLAGS),
+                  cygrpc.operation_receive_close_on_server(_EMPTY_FLAGS),
+                  cygrpc.operation_send_status_from_server(
+                      _common.EMPTY_METADATA,
+                      cygrpc.StatusCode.resource_exhausted,
+                      b'Max requests exceeded!', _EMPTY_FLAGS),)
+    rpc_state = _RPCState()
+    rpc_event.operation_call.start_server_batch(
+        operations, lambda ignored_event: (rpc_state, (),))
+    return rpc_state
+
+
+def _handle_with_method_handler(rpc_event, method_handler, thread_pool,
+                                server_state):
```

Comment:
What was an attractive property of this code before and that I'd like to preserve is that server state happens in the high-level functions located toward the end of the file and RPC state happens in the low-level functions located toward the beginning of the file and [there's currently only one painful exception](https://github.com/grpc/grpc/issues/6597). I think that rather than seeing this function take a server state, it should instead return a means by which the higher-level server-wide function can know that the RPC handled in this lower-level RPC-specific function has terminated. And the way for it to do that is... to return the [`concurrent.futures.Future`](https://docs.python.org/3/library/concurrent.futures.html#future-objects) that came from submitting the even-lower-level RPC-handling function to the server's thread pool (so this function would have two return values), so that _handle_call can just a few lines below where it has incremented `state.active_rpc_count` register a decrementing call with `add_done_callback`.